### ISSUE TITLE
fatal error: concurrent map read and map write

### ISSUE THREAD

**Issue by NotNotQuinn** [NotNotQuinn/go-websub#6]
```
fatal error: concurrent map read and map write

goroutine 63 [running]:
runtime.throw({0x59dfc5?, 0x50?})
        C:/Program Files/Go/src/runtime/panic.go:992 +0x76 fp=0xc00016fe60 sp=0xc00016fe30 pc=0x2f6156 
runtime.mapaccess1_faststr(0x54d180?, 0xc0001262c0?, {0xc00000c3e0, 0x1c})
        C:/Program Files/Go/src/runtime/map_faststr.go:22 +0x3a5 fp=0xc00016fec8 sp=0xc00016fe60 pc=0x2d2725
github.com/notnotquinn/go-websub.(*Hub).Publish(0xc000068b40, {0xc00000c3e0, 0x1c}, {0x596071, 0x10}, {0xc00000e550, 0x4a, 0x50})
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:540 +0x232 fp=0xc00016ff90 sp=0xc00016fec8 pc=0x5167f2
github.com/notnotquinn/go-websub.(*Hub).publishTopicUpdates.func1()
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:735 +0x46 fp=0xc00016ffe0 sp=0xc00016ff90 pc=0x518aa6
runtime.goexit()
        C:/Program Files/Go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc00016ffe8 sp=0xc00016ffe0 pc=0x322ae1
created by github.com/notnotquinn/go-websub.(*Hub).publishTopicUpdates
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:735 +0x50

goroutine 1 [sleep]:
time.Sleep(0x3b9aca00)
        C:/Program Files/Go/src/runtime/time.go:194 +0x133
main.main()
        D:/dev/websub-transport-system/test/main.go:39 +0xbf

goroutine 6 [chan receive]:
github.com/notnotquinn/go-websub.(*Hub).handleFailedPublishes(0xc000068b40)
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:644 +0x47
created by github.com/notnotquinn/go-websub.NewHub
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:124 +0x216

goroutine 7 [chan receive]:
github.com/notnotquinn/go-websub.(*Hub).removeExpiredSubscriptions(0xc000068b40, 0x0?)
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:671 +0x54
created by github.com/notnotquinn/go-websub.NewHub
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:125 +0x26a

goroutine 8 [chan receive]:
github.com/notnotquinn/go-websub.(*Hub).publishTopicUpdates(0xc000068b40)
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:725 +0x68
created by github.com/notnotquinn/go-websub.NewHub
        C:/Users/Qthie/go/pkg/mod/github.com/notnotquinn/go-websub@v0.2.1-0.20220331000917-79d9168a9b6b/hub.go:127 +0x2b6

goroutine 9 [IO wait]:
internal/poll.runtime_pollWait(0x1fdffdc4dd8, 0x72)
        C:/Program Files/Go/src/runtime/netpoll.go:302 +0x89
internal/poll.(*pollDesc).wait(0xc000037d40?, 0x0?, 0x0)
        C:/Program Files/Go/src/internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.execIO(0xc000098018, 0xc00013dbb8)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:175 +0xe5
internal/poll.(*FD).acceptOne(0xc000098000, 0x230, {0xc0002220f0?, 0x0?, 0x8?}, 0x1fdffbfffff?)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:942 +0x6d
internal/poll.(*FD).Accept(0xc000098000, 0xc00013dd90)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:976 +0x1d6
net.(*netFD).accept(0xc000098000)
        C:/Program Files/Go/src/net/fd_windows.go:139 +0x65
net.(*TCPListener).accept(0xc000094018)
        C:/Program Files/Go/src/net/tcpsock_posix.go:139 +0x28
net.(*TCPListener).Accept(0xc000094018)
        C:/Program Files/Go/src/net/tcpsock.go:288 +0x3d
net/http.(*Server).Serve(0xc000088000, {0x615a40, 0xc000094018})
        C:/Program Files/Go/src/net/http/server.go:3039 +0x385
net/http.(*Server).ListenAndServe(0xc000088000)
        C:/Program Files/Go/src/net/http/server.go:2968 +0x7d
net/http.ListenAndServe({0xc0000149c0, 0xe}, {0x613940?, 0xc000068b40})
        C:/Program Files/Go/src/net/http/server.go:3222 +0x96
created by main.startHub
        D:/dev/websub-transport-system/test/main.go:81 +0x32a

goroutine 10 [IO wait]:
internal/poll.runtime_pollWait(0x1fdffdc4ec8, 0x72)
        C:/Program Files/Go/src/runtime/netpoll.go:302 +0x89
internal/poll.(*pollDesc).wait(0xc0001421a0?, 0x0?, 0x0)
        C:/Program Files/Go/src/internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.execIO(0xc00020e018, 0xc00013fbb8)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:175 +0xe5
internal/poll.(*FD).acceptOne(0xc00020e000, 0x224, {0xc0000a80f0?, 0x0?, 0x8?}, 0x1fdffbfffff?)        
        C:/Program Files/Go/src/internal/poll/fd_windows.go:942 +0x6d
internal/poll.(*FD).Accept(0xc00020e000, 0xc00013fd90)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:976 +0x1d6
net.(*netFD).accept(0xc00020e000)
        C:/Program Files/Go/src/net/fd_windows.go:139 +0x65
net.(*TCPListener).accept(0xc000214000)
        C:/Program Files/Go/src/net/tcpsock_posix.go:139 +0x28
net.(*TCPListener).Accept(0xc000214000)
        C:/Program Files/Go/src/net/tcpsock.go:288 +0x3d
net/http.(*Server).Serve(0xc000206000, {0x615a40, 0xc000214000})
        C:/Program Files/Go/src/net/http/server.go:3039 +0x385
net/http.(*Server).ListenAndServe(0xc000206000)
        C:/Program Files/Go/src/net/http/server.go:2968 +0x7d
net/http.ListenAndServe({0xc000014a10, 0xe}, {0x6139a0?, 0xc000140000})
        C:/Program Files/Go/src/net/http/server.go:3222 +0x96
created by main.startNode
        D:/dev/websub-transport-system/test/main.go:52 +0x18a

goroutine 21 [IO wait]:
internal/poll.runtime_pollWait(0x1fdffdc4bf8, 0x72)
        C:/Program Files/Go/src/runtime/netpoll.go:302 +0x89
internal/poll.(*pollDesc).wait(0x325273?, 0x624527fc?, 0x0)
        C:/Program Files/Go/src/internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.execIO(0xc00020e518, 0x5c3868)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:175 +0xe5
internal/poll.(*FD).Read(0xc00020e500, {0xc000236000, 0x1000, 0x1000})
        C:/Program Files/Go/src/internal/poll/fd_windows.go:441 +0x25f
net.(*netFD).Read(0xc00020e500, {0xc000236000?, 0x2c5b90?, 0x4?})
        C:/Program Files/Go/src/net/fd_posix.go:55 +0x29
net.(*conn).Read(0xc00021a010, {0xc000236000?, 0x0?, 0x0?})
        C:/Program Files/Go/src/net/net.go:183 +0x45
net/http.(*persistConn).Read(0xc00010e6c0, {0xc000236000?, 0xc000228240?, 0xc000231d30?})
        C:/Program Files/Go/src/net/http/transport.go:1929 +0x4e
bufio.(*Reader).fill(0xc000234000)
        C:/Program Files/Go/src/bufio/bufio.go:106 +0x103
bufio.(*Reader).Peek(0xc000234000, 0x1)
        C:/Program Files/Go/src/bufio/bufio.go:144 +0x5d
net/http.(*persistConn).readLoop(0xc00010e6c0)
        C:/Program Files/Go/src/net/http/transport.go:2093 +0x1ac
created by net/http.(*Transport).dialConn
        C:/Program Files/Go/src/net/http/transport.go:1750 +0x173e

goroutine 23 [IO wait]:
internal/poll.runtime_pollWait(0x1fdffdc4928, 0x72)
        C:/Program Files/Go/src/runtime/netpoll.go:302 +0x89
internal/poll.(*pollDesc).wait(0x31f4d8?, 0x0?, 0x0)
        C:/Program Files/Go/src/internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.execIO(0xc00020e798, 0x5c3868)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:175 +0xe5
internal/poll.(*FD).Read(0xc00020e780, {0xc0000b4000, 0x1000, 0x1000})
        C:/Program Files/Go/src/internal/poll/fd_windows.go:441 +0x25f
net.(*netFD).Read(0xc00020e780, {0xc0000b4000?, 0x3708e6?, 0x0?})
        C:/Program Files/Go/src/net/fd_posix.go:55 +0x29
net.(*conn).Read(0xc00021a018, {0xc0000b4000?, 0x0?, 0x0?})
        C:/Program Files/Go/src/net/net.go:183 +0x45
net/http.(*connReader).Read(0xc00008c240, {0xc0000b4000, 0x1000, 0x1000})
        C:/Program Files/Go/src/net/http/server.go:780 +0x16d
bufio.(*Reader).fill(0xc000086240)
        C:/Program Files/Go/src/bufio/bufio.go:106 +0x103
bufio.(*Reader).Peek(0xc000086240, 0x4)
        C:/Program Files/Go/src/bufio/bufio.go:144 +0x5d
net/http.(*conn).readRequest(0xc00023a000, {0x615cf0, 0xc00009e040})
        C:/Program Files/Go/src/net/http/server.go:985 +0x1f8
net/http.(*conn).serve(0xc00023a000, {0x615d98, 0xc00008c150})
        C:/Program Files/Go/src/net/http/server.go:1891 +0x32b
created by net/http.(*Server).Serve
        C:/Program Files/Go/src/net/http/server.go:3071 +0x4db

goroutine 22 [select]:
net/http.(*persistConn).writeLoop(0xc00010e6c0)
        C:/Program Files/Go/src/net/http/transport.go:2392 +0xf5
created by net/http.(*Transport).dialConn
        C:/Program Files/Go/src/net/http/transport.go:1751 +0x1791

goroutine 34 [IO wait]:
internal/poll.runtime_pollWait(0x1fdffdc4b08, 0x72)
        C:/Program Files/Go/src/runtime/netpoll.go:302 +0x89
internal/poll.(*pollDesc).wait(0x34d5b3?, 0x615ad0?, 0x0)
        C:/Program Files/Go/src/internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.execIO(0xc000098298, 0x5c3868)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:175 +0xe5
internal/poll.(*FD).Read(0xc000098280, {0xc00014e000, 0x1000, 0x1000})
        C:/Program Files/Go/src/internal/poll/fd_windows.go:441 +0x25f
net.(*netFD).Read(0xc000098280, {0xc00014e000?, 0x0?, 0xc0000b16c8?})
        C:/Program Files/Go/src/net/fd_posix.go:55 +0x29
net.(*conn).Read(0xc0000a0008, {0xc00014e000?, 0xc0000841a0?, 0xc0000b16f8?})
        C:/Program Files/Go/src/net/net.go:183 +0x45
net/http.(*connReader).Read(0xc00007d380, {0xc00014e000, 0x1000, 0x1000})
        C:/Program Files/Go/src/net/http/server.go:780 +0x16d
bufio.(*Reader).fill(0xc0000526c0)
        C:/Program Files/Go/src/bufio/bufio.go:106 +0x103
bufio.(*Reader).ReadSlice(0xc0000526c0, 0x0?)
        C:/Program Files/Go/src/bufio/bufio.go:371 +0x2f
bufio.(*Reader).ReadLine(0xc0000526c0)
        C:/Program Files/Go/src/bufio/bufio.go:400 +0x27
net/textproto.(*Reader).readLineSlice(0xc00020a270)
        C:/Program Files/Go/src/net/textproto/reader.go:57 +0x99
net/textproto.(*Reader).ReadLine(...)
        C:/Program Files/Go/src/net/textproto/reader.go:38
net/http.readRequest(0xc0000a0008?)
        C:/Program Files/Go/src/net/http/request.go:1029 +0x79
net/http.(*conn).readRequest(0xc0000aa000, {0x615cf0, 0xc0000503c0})
        C:/Program Files/Go/src/net/http/server.go:988 +0x24a
net/http.(*conn).serve(0xc0000aa000, {0x615d98, 0xc00020a150})
        C:/Program Files/Go/src/net/http/server.go:1891 +0x32b
created by net/http.(*Server).Serve
        C:/Program Files/Go/src/net/http/server.go:3071 +0x4db

goroutine 56 [IO wait]:
internal/poll.runtime_pollWait(0x1fdffdc4a18, 0x72)
        C:/Program Files/Go/src/runtime/netpoll.go:302 +0x89
        C:/Program Files/Go/src/internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.execIO(0xc00007f198, 0x5c3868)
        C:/Program Files/Go/src/internal/poll/fd_windows.go:175 +0xe5
internal/poll.(*FD).Read(0xc00007f180, {0xc000162000, 0x1000, 0x1000})
        C:/Program Files/Go/src/internal/poll/fd_windows.go:441 +0x25f
net.(*netFD).Read(0xc00007f180, {0xc000162000?, 0x0?, 0x4?})
net.(*conn).Read(0xc000006090, {0xc000162000?, 0xc000006080?, 0x1?})
        C:/Program Files/Go/src/net/net.go:183 +0x45
net/http.(*persistConn).Read(0xc00010e7e0, {0xc000162000?, 0x307920?, 0xc0000afec8?})
        C:/Program Files/Go/src/net/http/transport.go:1929 +0x4e
bufio.(*Reader).fill(0xc000052b40)
        C:/Program Files/Go/src/bufio/bufio.go:106 +0x103
bufio.(*Reader).Peek(0xc000052b40, 0x1)
        C:/Program Files/Go/src/bufio/bufio.go:144 +0x5d
net/http.(*persistConn).readLoop(0xc00010e7e0)
        C:/Program Files/Go/src/net/http/transport.go:2093 +0x1ac
created by net/http.(*Transport).dialConn
        C:/Program Files/Go/src/net/http/transport.go:1750 +0x173e

goroutine 57 [select]:
net/http.(*persistConn).writeLoop(0xc00010e7e0)
        C:/Program Files/Go/src/net/http/transport.go:2392 +0xf5
created by net/http.(*Transport).dialConn
        C:/Program Files/Go/src/net/http/transport.go:1751 +0x1791
```

**Comment by NotNotQuinn** [comment]
dank stuff

**Comment by NotNotQuinn** [comment]
Should be fixed in 7b507864cf6d80c8573eb1dfd5fdbd3ac86419f8?
